src/persist-client/src/lib.rs

```diff
@@ -491,33 +495,44 @@ impl PersistClient {
     /// If `shard_id` has never been used before, initializes a new shard and
     /// returns handles with `since` and `upper` frontiers set to initial values
     /// of `Antichain::from_elem(T::minimum())`.
+    ///
+    /// The `schema` parameter is currently unused, but should be an object
+    /// that represents the schema of the data in the shard. This will be required
+    /// in the future.
     #[instrument(level = "debug", skip_all, fields(shard = %shard_id))]
-    pub async fn open<K, V, T, D>(
+    pub async fn open<K, V, T, D, S>(
         &self,
         shard_id: ShardId,
         purpose: &str,
+        schema: S,
     ) -> Result<(WriteHandle<K, V, T, D>, ReadHandle<K, V, T, D>), InvalidUsage<T>>
     where
         K: Debug + Codec,
         V: Debug + Codec,
         T: Timestamp + Lattice + Codec64,
         D: Semigroup + Codec64 + Send + Sync,
+        S: Clone,
     {
         Ok((
-            self.open_writer(shard_id, purpose).await?,
-            self.open_leased_reader(shard_id, purpose).await?,
+            self.open_writer(shard_id, purpose, schema.clone()).await?,
+            self.open_leased_reader(shard_id, purpose, schema).await?,
         ))
     }
 
-    /// [Self::open], but returning only a [ReadHandle].
+    /// [Self::open], but returning only a [/eadHandle].
```

Comment:
oh oh?

Reply:
RIP, will have to fix this in a followup